bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/converter/property/UnixDateConverter.java

```diff
@@ -8,26 +8,27 @@
  */
 package org.openhab.binding.weather.internal.converter.property;
 
-import java.util.Calendar;
-
 import org.openhab.binding.weather.internal.converter.Converter;
 import org.openhab.binding.weather.internal.converter.ConverterType;
 
+import java.util.Calendar;
+
+
 /**
  * Date converter for a unix timestamp.
  *
  * @author Gerhard Riegler
  * @since 1.6.0
  */
 public class UnixDateConverter implements Converter<Calendar> {
-
     /**
      * {@inheritDoc}
      */
     @Override
     public Calendar convert(String value) {
         Calendar cal = Calendar.getInstance();
-        cal.setTimeInMillis(Long.valueOf(value) * 1000);
+        cal.setTimeInMillis((long) (Double.valueOf(value) * 1000));
```

Comment:
What's the reason for this?

Reply:
Weatherbit provides a unix timstamp with fractional miliseconds
like "1547066713.2043"